Fix upgrade documentation about form handling (#4330)
marmelab/react-admin: UPGRADE.md

        }

        create(
            {

>                 data: { ...formState.values, average_note: 10 },

Reviewer: Shouldn't it be the following ?

```suggestion
                payload: { data: { ...formState.values, average_note: 10 }, },
```

Author: right

};
```

## Custom Forms using `reduxForm()` must be replaced by using `<Form>`

[Final form migration documentation here](https://final-form.org/docs/react-final-form/migration/redux-form) explains the various changes you have to perform in your code.

```diff
-import { reduxForm } from 'redux-form'
+import { Form } from 'react-final-form'

-const CustomForm = reduxForm({ form: 'record-form', someOptions: true })(({ record, resource }) => (	
+const CustomForm = ({ record, resource }) => (

> +   <Form someOptions={true}>

Reviewer: ```suggestion
+    <Form someOptions={true}>
+        {({ handleSubmit }) => (
```

>     <div>	    

Reviewer: ```suggestion
    +            <form onSubmit={handleSubmit}>
```

        <Typography>Notes</Typography>
        <TextInput source="note" />

>     </div>

Reviewer: ```suggestion
    </form>
```

> +   </Form>

Reviewer: ```suggestion
+        )}
+    </Form>
```

Reviewer: The whole snippet might need a little reformatting

};
```

> ## Custom Forms using `reduxForm()` must be replaced by using `<Form>`

Reviewer 2: ```suggestion
## Custom Forms Using `reduxForm()` Must Be Replaced By The `<Form>` Component
```

> [Final form migration documentation here](https://final-form.org/docs/react-final-form/migration/redux-form) explains the various changes you have to perform in your code.

Reviewer 2: ```suggestion
The [final-form migration documentation here](https://final-form.org/docs/react-final-form/migration/redux-form) explains the various changes you have to perform in your code.
```

```diff
-import { reduxForm } from 'redux-form'
+import { Form } from 'react-final-form'

-const CustomForm = reduxForm({ form: 'record-form', someOptions: true })(({ record, resource }) => (
+const CustomForm = ({ record, resource }) => (
+    <Form someOptions={true}>
+        {({ handleSubmit }) => (
+            <form onSubmit={handleSubmit}>
-            <Fragment>
                <Typography>Notes</Typography>
                <TextInput source="note" />
+            </form>
-            </Fragment>
+        )}
+    </Form>
+);
-));
```

> ## Material-ui icons have changed

Reviewer 2: ```suggestion
## Material-ui Icons Have Changed
```

> If you were using Material-ui icons for your design, just take care that the `v4.0` have removed some existing icons since the `v1.5`.

Reviewer 2: ```suggestion
If you were using Material-ui icons for your design, be aware that some icons present in 1.X versions were removed from version 4.0.
```

Example: 

> `LightbulbOutline` is no more available in `@Material-ui/icons`

Reviewer 2: Add line breaks for better readbility
